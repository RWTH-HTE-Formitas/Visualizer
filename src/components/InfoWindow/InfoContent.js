import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import SwipeableViews from 'react-swipeable-views';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Moment from 'react-moment';

function TabContainer({ children, dir }) {
  return (
    <Typography component="div" dir={dir} style={{ padding: 3 * 2 }}>
      {children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired,
  dir: PropTypes.string.isRequired,
};

const styles = theme => ({
  root: {
    backgroundColor: theme.palette.background.paper,
    width: '100%',
  },
});

class InfoContent extends React.Component {

  state = {
    value: 0,
  };

  handleChange = (event, value) => {
    this.setState({ value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  showDetail(type, source) {

  }

  render() {
    const { classes, theme } = this.props;

    const showWindow = this.props.data.showWindow;
    const inpData = this.props.data.objectData;

    let textNotes = [];
    let text0 = "";
    let text1 = "";

    //this.setState({rawtexts: rawtexts})



    let audioNotes = [];
    let imageNotes = [];


    if (inpData != null && inpData.Notes != null) {
      Object.values(inpData.Notes).map((note, i) => {
        if (note.TextNotes != null) {


            var link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.txt';
            fetch(link).then(response => {response.text().then(text => {
                if(i == 0)
                  this.setState({text0: text})
                if(i == 1)
                  this.setState({text1: text})
            })});



          textNotes = textNotes.concat(Object.values(note.TextNotes));
        }
        if (note.AudioNotes != null) {
          audioNotes = audioNotes.concat(Object.values(note.AudioNotes));

          /*
          let link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.mp3';
          fetch(link).then(response => {response.text().then(text => {
              if(i == 0)
                this.setState({text0: text})
              if(i == 1)
                this.setState({text1: text})
          })});
          */
        }
        if (note.ImageNotes != null) {
          imageNotes = imageNotes.concat(Object.values(note.ImageNotes));
        }
      })
    }



    return (
      <div className={classes.root}>
        <AppBar position="static" color="default">
          <Tabs
            value={this.state.value}
            onChange={this.handleChange}
            indicatorColor="primary"
            textColor="primary"
            fullWidth
          >
            <Tab label="Info" />
            <Tab label="Text" />
            <Tab label="Picture" />
            <Tab label="Voice" />
          </Tabs>
        </AppBar>
        <SwipeableViews
          axis={theme.direction === 'rtl' ? 'x-reverse' : 'x'}
          index={this.state.value}
          onChangeIndex={this.handleChangeIndex}
        >
          <TabContainer dir={theme.direction}>
            <table className="table table-text">
              <tbody>
                {
                  Object.keys(inpData).map((k, i) => {
                    const val = inpData[k]
                    if (typeof (val) == 'string') {
                      return (
                        <tr key={i}>
                          <td>{k}: </td>
                          <td>{val}</td>
                        </tr>
                      );
                    }
                  })
                }
              </tbody>
            </table>
          </TabContainer>
          <TabContainer dir={theme.direction}><table className="table table-text">
            <tbody>
              {
                textNotes.map((data, i) => {
                  var item = data.URL.split('/').slice(-1)[0]
                  var link = data.URL;
                  /*
                  link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.txt';
                  fetch(link).then(response => {response.text().then(text => {
                    this.setState({rawtexts: rawtexts})
                    console.log(this.state.test)
                  })
                  })
                  */

                  var dateRaw = new Date(data.Date * 1000);
                  return (
                    <tr key={i}>
                      <td className="text">
                      {
                        this.state.text0
                      }

                      </td>
                      <td className="date">
                        <span><Moment format="MMM DD, YYYY">{dateRaw}</Moment></span>
                        <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                      </td>
                    </tr>
                  );
                })
              }
            </tbody>
          </table>
          </TabContainer>
          <TabContainer dir={theme.direction}><table className="table table-picture">
            <tbody>
              {
                imageNotes.map((data, i) => {
                  var link = data.URL
                  var dateRaw = new Date(data.Date * 1000)
                  // override
                  link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.jpg';
                  return (
                    <tr key={i}>
                      <td className="text">
                        <a href="#" onClick={e => this.showDetail('image', link)}>
                          <img src={link} />
                        </a>
                      </td>
                      <td className="date">
                        <span><Moment format="MMM DD, YYYY">{dateRaw}</Moment></span>
                        <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                      </td>
                    </tr>
                  );
                })
              }
            </tbody>
          </table>
          </TabContainer>
          <TabContainer dir={theme.direction}><table className="table table-audio">
            <tbody>
              {
                audioNotes.map((data, i) => {
                  var item = data.URL.split('/').slice(-1)[0]
                  var link = data.URL
                  var dateRaw = new Date(data.Date * 1000)
                  return (
                    <tr key={i}>
                      <td className="text">
                      <audio controls>
                          <source src="https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.mp3" type="audio/mpeg" />
                        Your browser does not support the audio element.
                      </audio>
                      </td>
                      <td className="date">
                        <span><Moment format="MMM DD, YYYY">{dateRaw}</Moment></span>
                        <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                      </td>
                    </tr>
                  );
                })
              }
            </tbody>
          </table>
          </TabContainer>

        </SwipeableViews>
      </div>
    );
  }
}

InfoContent.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(InfoContent);
