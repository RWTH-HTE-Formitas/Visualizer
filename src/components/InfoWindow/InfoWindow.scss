.Modal {
  height: 100%;
  
}

.Overlay {
  position: fixed;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba($color: #000000, $alpha: 0.5);
  z-index: 199;
}

.header {
  width: 100%;
  height: 5%;
  
  button {
    position: absolute;
    right: 10px;
    top: 10px;
  }
}

.frame {
  width: 100%;
  height: 95%;
  text-align: center;

  .helper {
    display: inline-block;
    height: 100%;
    vertical-align: middle;
  }
  
  img {
    margin:0 auto;
    vertical-align: middle;
  }

  audio {
    width: 600px;
  }
  
  textarea {
    width: 400px;
    height: 200px;
  }
}
