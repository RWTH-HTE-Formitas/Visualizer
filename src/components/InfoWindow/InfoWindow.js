
import React from 'react';
import SwipeableViews from 'react-swipeable-views';
import Typography from '@material-ui/core/Typography';
import Moment from 'react-moment';
import { AppBar, Tabs, Tab, Button } from '@material-ui/core';
import { Table, TableRow, TableCell, TableBody } from '@material-ui/core';
import ReactModal from 'react-modal';
import './InfoWindow.scss'

function TabContainer({children}) {

  return (
    <Typography component="div" style={{padding: 3 * 2}}>
      {children}
    </Typography>
  );
}

class InfoWindow extends React.Component {

  constructor(props) {

    super(props);

    this.state = {
      activeTab: 0,
      showModal: false
    };
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
    this.showImage = this.showImage.bind(this);
  }

  detail = null;

  onChangeTab(newTab) {

    this.setState({
      activeTab: newTab
    });
  };

  showImage(url) {
    this.detail = <img src={url} alt="" />;
    this.handleOpenModal();
  }

  handleOpenModal() {
    this.setState({ showModal: true });
  }

  handleCloseModal() {
    this.setState({ showModal: false });
  }

  componentWillReceiveProps(nextProps) {
    // set active tab to 0 again if new data comes
    this.setState({
      activeTab: 0
    });
  }

  render() {

    const object = this.props.selectedAnnotatedObject;

    let textNotes = [];
    let audioNotes = [];
    let imageNotes = [];

    if (object != null && object.Notes != null) {

      Object.values(object.Notes).forEach((note, i) => {

        if (note.TextNotes != null) {

          var link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.txt';

          fetch(link).then(response => {

            response.text().then(text => {

              if (i === 0) {

                this.setState({text0: text})
              }
              else if (i === 1) {

                this.setState({text1: text})
              }
            })
          });
          Object.values(note.TextNotes).forEach(
            (data) => {
              const obj = data;
              obj.email = note.Email;
              textNotes.push(obj);
            }
          );
        }

        if (note.AudioNotes != null) {
          Object.values(note.AudioNotes).forEach(
            (data) => {
              const obj = data;
              obj.email = note.Email;
              audioNotes.push(obj);
            }
          );
        }

        if (note.ImageNotes != null) {
          Object.values(note.ImageNotes).forEach(
            (data) => {
              const obj = data;
              obj.email = note.Email;
              imageNotes.push(obj);
            }
          );
        }
      });
    }

    return (
      <div style={ object ? {} : { display: 'none'} }>
        <AppBar position="static" color="default">
          <Tabs
            value={this.state.activeTab}
            onChange={(_, index) => this.onChangeTab(index)}
            indicatorColor="primary"
            textColor="primary"
            variant="fullWidth"
          >
            <Tab label="Info"/>
            <Tab label="Text"/>
            <Tab label="Picture"/>
            <Tab label="Voice"/>
          </Tabs>
        </AppBar>
        <SwipeableViews
          index={this.state.activeTab}
          onChangeIndex={(index) => this.onChangeTab(index)}
        >
          <TabContainer>
          <Table>
            <TableBody>
              {
                Object.keys(object ? object : {}).map((k, i) => {

                  const val = object[k];

                  if (typeof (val) == 'string') {
                    return (
                      <TableRow key={i}>
                        <TableCell component="th" scope="row" align="right"><Typography variant="body1">{k}:</Typography></TableCell>
                        <TableCell><Typography variant="body1">{val}</Typography></TableCell>
                      </TableRow>
                    );
                  }
                  else {
                    return (null);
                  }
                })
              }
            </TableBody>
          </Table>
          </TabContainer>
          <TabContainer>
            <Table>
              <TableBody>
              {
                textNotes.map((data, i) => {

                  const dateRaw = new Date(data.Date * 1000);

                  return (
                    <TableRow key={i}>
                      <TableCell className="text">
                        <Typography variant="body1">{this.state.text0}</Typography>
                      </TableCell>
                      <TableCell>
                        <Typography variant="body1">{data.email}</Typography>
                      </TableCell>
                      <TableCell className="date">
                        <Typography variant="body1">
                          <Moment format="MMM DD, YYYY">{dateRaw}</Moment>
                          <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                        </Typography>
                      </TableCell>
                    </TableRow>
                  );
                })
              }
              </TableBody>
            </Table>
          </TabContainer>
          <TabContainer>
            <Table>
              <TableBody>
              {
                imageNotes.map((data, i) => {

                  //const link = data.URL;
                  const dateRaw = new Date(data.Date * 1000);

                  // override
                  const link = 'https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.jpg';

                  return (
                    <TableRow key={i}>
                      <TableCell className="text">
                        <img src={link} style={{maxWidth: 100}}  onClick={() => this.showImage(link)} alt=""/>
                      </TableCell>
                      <TableCell>
                        <Typography variant="body1">{data.email}</Typography>
                      </TableCell>
                      <TableCell className="date">
                        <Typography variant="body1">
                          <Moment format="MMM DD, YYYY">{dateRaw}</Moment>
                          <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                        </Typography>
                      </TableCell>
                    </TableRow>
                  );
                })
              }
              </TableBody>
            </Table>
          </TabContainer>
          <TabContainer>
            <Table>
              <TableBody>
              {
                audioNotes.map((data, i) => {

                  const dateRaw = new Date(data.Date * 1000);

                  return (
                    <TableRow key={i}>
                      <TableCell className="text">
                        <audio controls>
                          <source src="https://raw.githubusercontent.com/RWTH-HTE-Formitas/Visualizer/tmp/sample.mp3"
                                  type="audio/mpeg"/>
                          Your browser does not support the audio element.
                        </audio>
                      </TableCell>
                      <TableCell>
                        <Typography variant="body1">{data.email}</Typography>
                      </TableCell>
                      <TableCell className="date">
                        <Typography variant="body1">
                          <Moment format="MMM DD, YYYY">{dateRaw}</Moment>
                          <b> <Moment format="HH:mm">{dateRaw}</Moment></b>
                        </Typography>
                      </TableCell>
                    </TableRow>
                  );
                })
              }
              </TableBody>
            </Table>
          </TabContainer>
        </SwipeableViews>
        <ReactModal
          isOpen={this.state.showModal}
          contentLabel="onRequestClose Example"
          onRequestClose={this.handleCloseModal}
          className="Modal"
          overlayClassName="Overlay"
        >
          <div className="header">
              <Button onClick={this.handleCloseModal} variant="contained" size="small" color="secondary">X</Button>
          </div>
          <div className="body frame">
              <span className="helper"></span>
              {this.detail}
          </div>
        </ReactModal>
      </div>
    );
  }
}

export default (InfoWindow);
