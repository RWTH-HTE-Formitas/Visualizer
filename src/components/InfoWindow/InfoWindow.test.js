import React from 'react';
import Enzyme, {shallow} from 'enzyme' ;
import Adapter from 'enzyme-adapter-react-16';
import InfoWindow from './InfoWindow';

Enzyme.configure({adapter: new Adapter()});

describe("InfoWindow snapshot testing", () => {
  test('Renders correctly', () => {
    const wrapper = shallow(<InfoWindow />);
    expect(wrapper).toMatchSnapshot();
});
});