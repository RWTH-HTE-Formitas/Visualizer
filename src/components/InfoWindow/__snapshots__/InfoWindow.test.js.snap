// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoWindow component rendering render a label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoWindow />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(AppBar)
          color="default"
          position="static"
        >
          <WithStyles(Tabs)
            indicatorColor="primary"
            onChange={[Function]}
            textColor="primary"
            value={0}
            variant="fullWidth"
          >
            <WithStyles(Tab)
              label="Info"
            />
            <WithStyles(Tab)
              label="Text"
            />
            <WithStyles(Tab)
              label="Picture"
            />
            <WithStyles(Tab)
              label="Voice"
            />
          </WithStyles(Tabs)>
        </WithStyles(AppBar)>,
        <ReactSwipableView
          animateHeight={false}
          animateTransitions={true}
          axis="x"
          disableLazyLoading={false}
          disabled={false}
          enableMouseEvents={false}
          hysteresis={0.6}
          ignoreNativeScroll={false}
          index={0}
          onChangeIndex={[Function]}
          resistance={false}
          springConfig={
            Object {
              "delay": "0s",
              "duration": "0.35s",
              "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
            }
          }
          threshold={5}
        >
          <TabContainer>
            <WithStyles(Table)>
              <WithStyles(TableBody) />
            </WithStyles(Table)>
          </TabContainer>
          <TabContainer>
            <WithStyles(Table)>
              <WithStyles(TableBody) />
            </WithStyles(Table)>
          </TabContainer>
          <TabContainer>
            <WithStyles(Table)>
              <WithStyles(TableBody) />
            </WithStyles(Table)>
          </TabContainer>
          <TabContainer>
            <WithStyles(Table)>
              <WithStyles(TableBody) />
            </WithStyles(Table)>
          </TabContainer>
        </ReactSwipableView>,
      ],
      "style": Object {
        "display": "none",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Tabs)
            indicatorColor="primary"
            onChange={[Function]}
            textColor="primary"
            value={0}
            variant="fullWidth"
          >
            <WithStyles(Tab)
              label="Info"
            />
            <WithStyles(Tab)
              label="Text"
            />
            <WithStyles(Tab)
              label="Picture"
            />
            <WithStyles(Tab)
              label="Voice"
            />
          </WithStyles(Tabs)>,
          "color": "default",
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Tab)
                label="Info"
              />,
              <WithStyles(Tab)
                label="Text"
              />,
              <WithStyles(Tab)
                label="Picture"
              />,
              <WithStyles(Tab)
                label="Voice"
              />,
            ],
            "indicatorColor": "primary",
            "onChange": [Function],
            "textColor": "primary",
            "value": 0,
            "variant": "fullWidth",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Picture",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Voice",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animateHeight": false,
          "animateTransitions": true,
          "axis": "x",
          "children": Array [
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>,
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>,
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>,
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>,
          ],
          "disableLazyLoading": false,
          "disabled": false,
          "enableMouseEvents": false,
          "hysteresis": 0.6,
          "ignoreNativeScroll": false,
          "index": 0,
          "onChangeIndex": [Function],
          "resistance": false,
          "springConfig": Object {
            "delay": "0s",
            "duration": "0.35s",
            "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
          },
          "threshold": 5,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableBody) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableBody) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableBody) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableBody) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)
            color="default"
            position="static"
          >
            <WithStyles(Tabs)
              indicatorColor="primary"
              onChange={[Function]}
              textColor="primary"
              value={0}
              variant="fullWidth"
            >
              <WithStyles(Tab)
                label="Info"
              />
              <WithStyles(Tab)
                label="Text"
              />
              <WithStyles(Tab)
                label="Picture"
              />
              <WithStyles(Tab)
                label="Voice"
              />
            </WithStyles(Tabs)>
          </WithStyles(AppBar)>,
          <ReactSwipableView
            animateHeight={false}
            animateTransitions={true}
            axis="x"
            disableLazyLoading={false}
            disabled={false}
            enableMouseEvents={false}
            hysteresis={0.6}
            ignoreNativeScroll={false}
            index={0}
            onChangeIndex={[Function]}
            resistance={false}
            springConfig={
              Object {
                "delay": "0s",
                "duration": "0.35s",
                "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
              }
            }
            threshold={5}
          >
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>
            <TabContainer>
              <WithStyles(Table)>
                <WithStyles(TableBody) />
              </WithStyles(Table)>
            </TabContainer>
          </ReactSwipableView>,
        ],
        "style": Object {
          "display": "none",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tabs)
              indicatorColor="primary"
              onChange={[Function]}
              textColor="primary"
              value={0}
              variant="fullWidth"
            >
              <WithStyles(Tab)
                label="Info"
              />
              <WithStyles(Tab)
                label="Text"
              />
              <WithStyles(Tab)
                label="Picture"
              />
              <WithStyles(Tab)
                label="Voice"
              />
            </WithStyles(Tabs)>,
            "color": "default",
            "position": "static",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Tab)
                  label="Info"
                />,
                <WithStyles(Tab)
                  label="Text"
                />,
                <WithStyles(Tab)
                  label="Picture"
                />,
                <WithStyles(Tab)
                  label="Voice"
                />,
              ],
              "indicatorColor": "primary",
              "onChange": [Function],
              "textColor": "primary",
              "value": 0,
              "variant": "fullWidth",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Info",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Picture",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Voice",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animateHeight": false,
            "animateTransitions": true,
            "axis": "x",
            "children": Array [
              <TabContainer>
                <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>
              </TabContainer>,
              <TabContainer>
                <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>
              </TabContainer>,
              <TabContainer>
                <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>
              </TabContainer>,
              <TabContainer>
                <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>
              </TabContainer>,
            ],
            "disableLazyLoading": false,
            "disabled": false,
            "enableMouseEvents": false,
            "hysteresis": 0.6,
            "ignoreNativeScroll": false,
            "index": 0,
            "onChangeIndex": [Function],
            "resistance": false,
            "springConfig": Object {
              "delay": "0s",
              "duration": "0.35s",
              "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
            },
            "threshold": 5,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(TableBody) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(TableBody) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(TableBody) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <WithStyles(Table)>
                  <WithStyles(TableBody) />
                </WithStyles(Table)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(TableBody) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
